cmd/frontend/graphqlbackend/external_service.go

```
@@ -74,6 +82,34 @@ func (r *externalServiceResolver) UpdatedAt() DateTime {
 	return DateTime{Time: r.externalService.UpdatedAt}
 }
 
+func (r *externalServiceResolver) CampaignWebhookURL() *string {
+	r.webhookURLOnce.Do(func() {
+		parsed, err := extsvc.ParseConfig(r.externalService.Kind, r.externalService.Config)
+		if err != nil {
```

Why not return this error? We use the same pattern (`sync.Once`, value field, error field) in other resolvers. Example:

https://github.com/sourcegraph/sourcegraph/blob/91a98b8aad34aa5642222088efaa45756b237865/cmd/frontend/graphqlbackend/repository_comparison.go#L225-L226

In this case the conversion would not be too hard and it would lead to one less error being swallowed somewhere :)

---

Ping on this @ryanslade :)

---

Ignore the ping, sorry. I had the wrong view open in GitHub.